### ISSUE TITLE
accessibility

### ISSUE THREAD

**Issue by jquense** [react-bootstrap/react-overlays#12]
There are a few issues here, that are not trivial to fix, to make the modal actually useful to a SR and not just what upstream bootstrap shows as accessible.

This is really made complicated by both stacking, and multiple containers.

1. when a modal is open `aria-hidden` should be applied to the rest of the document content, so a SR can't shift virtual focus to it (which is separate from actual focus which is already handled). Essentially we need to create a virtual backdrop for the SR, so that it can't interact with anything else. This only actually makes sense tho when the container is the document body, or some other top level root. If the container is some deeper component, adding aria-hidden to it only hides some of the content.

2. On top of this, any additional modals _with backdrops_ that open should cause existing modals to be aria-hidden.

This gets sort of messy pretty quick. I'd like to handle it if we can, since doing this correctly is not super easy for a consumer, but I don't want to add a ton of overhead either.

**Comment by jquense** [comment]
I think we can simplify this by just assuming that when we say "modal" we actually mean modal, ie. you can't interact with anything behind it
